test/test_fx.py

```diff
@@ -603,6 +604,32 @@ def test_construct_root_dict(self):
         ref_out : torch.Tensor = linear_mod(x) + add_param
         self.assertEqual(out, ref_out)
 
-
+    def test_get_all_users_of(self):
+        def find_use(arg: Any) -> bool:
+            if isinstance(arg, (tuple, list)):
+                return any(find_use(elem) for elem in arg)
+            elif isinstance(arg, dict):
+                return any(find_use(v) for k, v in arg.items())
+            elif isinstance(arg, slice):
+                return any(find_use(arg.start), find_use(arg.stop), find_use(arg.step))
+            elif isinstance(arg, Node):
+                return arg is node
+            else:
+                return False
```

Comment:
If you have to copy over the implementation of the function you're testing, the test is not a good test. Rather than trying to programmatically generate verification here, I suggest manually specifying the nodes you expect to find, for example:

```
        expected_uses : Dict[Node, Set[Node]] = {
          a : set([b]),
          b : set([c]),
          <etc...>
        }
        for i, node in enumerate(graph.nodes):
            user_indexes = GraphManipulation.get_all_users_of(gm, i)
            for expected_use in expected_uses[node]:
              assert expected_use in user_indexes
```